packages/webapp/src/pages/induction/[id].tsx

```diff
@@ -1,128 +1,106 @@
-import { useMemo, useState } from "react";
+import React, { useMemo } from "react";
 import { useRouter } from "next/router";
 
 import {
     CallToAction,
+    Card,
     RawLayout,
     SingleColLayout,
-    useFetchedData,
     useIsCommunityActive,
-    useUALAccount,
 } from "_app";
 import {
-    getInductionWithEndorsements,
-    Induction,
-    InductionStepEndorsement,
-    InductionStepProfile,
-    InductionStepVideo,
-    InductionStatus,
     getInductionStatus,
-    Endorsement,
+    InductionRole,
+    InductionStatus,
+    InviteeJourney,
+    InviterWitnessJourney,
+    ThirdPartyJourney,
+    useGetInductionWithEndorsements,
+    useInductionUserRole,
 } from "inductions";
 
 export const InductionDetailsPage = () => {
     const router = useRouter();
     const inductionId = router.query.id;
-    const [ualAccount] = useUALAccount();
 
-    const [reviewStep, setReviewStep] = useState<
-        "profile" | "video" | undefined
-    >();
+    const { isLoading: isLoadingCommunityState } = useIsCommunityActive();
 
     const {
-        data: isCommunityActive,
-        isLoading: isLoadingCommunityState,
-    } = useIsCommunityActive();
-
-    const [inductionEndorsements, isLoadingEndorsements] = useFetchedData<{
-        induction: Induction;
-        endorsements: Endorsement[];
-    }>(getInductionWithEndorsements, inductionId);
-
-    const induction = inductionEndorsements
-        ? inductionEndorsements.induction
-        : undefined;
+        data,
+        isLoading: isLoadingEndorsements,
+    } = useGetInductionWithEndorsements(inductionId as string);
 
-    const endorsements = inductionEndorsements
-        ? inductionEndorsements.endorsements
-        : [];
+    const isLoading = isLoadingEndorsements || isLoadingCommunityState;
 
-    // TODO: Consider deriving the user's role here and return an enum of roles: INVITER, ENDORSER, INVITEE, MEMBER, EOS_USER, UNAUTHENTICATED.
-    // Almost every child component of this page cares about the role of the user in relation to the invite/induction. We can pass role down.
-    const isEndorser = useMemo(() => 
-         endorsements.some(
-            (endorsement) => endorsement.endorser === ualAccount?.accountName
-        )
-    , [ualAccount, endorsements]);
+    const induction = data?.induction;
+    const endorsements = data?.endorsements ?? [];
 
-    const status = getInductionStatus(induction);
+    const userRole = useInductionUserRole(endorsements, induction);
+    const status = getInductionStatus(induction, endorsements);
 
-    const renderInductionStep = useMemo(() => {
+    const renderInductionJourney = useMemo(() => {
```

Comment:
I think it's a good opportunity to extract this whole memo to another component container... this page component would be way cleaner

Reply:
Good call.